app/main.py

```diff
@@ -1 +1,69 @@
-# write your code here
+class SoftwareEngineer:
+    def __init__(self, name):
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill):
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name):
+        super(FrontendDeveloper, self).__init__(name)
+        self.skills = [
+            "JavaScript",
+            "HTML",
+            "CSS"
+        ]
+
+    def create_awesome_web_page(self):
+        print(f"{self.name} is creating a webpage...")
+
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(self, name):
+        super(BackendDeveloper, self).__init__(name)
+        self.skills = [
+            "Python",
+            "SQL",
+            "Django"
+        ]
+
+    def create_powerful_api(self):
+        print(f"{self.name} is creating an API...")
+
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    def __init__(self, name):
+        super(AndroidDeveloper, self).__init__(name)
+        self.skills = [
+            "Java",
+            "Android studio"
+        ]
+
+    def create_smooth_mobile_app(self):
+        print(f"{self.name} is creating a mobile app...")
+
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(FrontendDeveloper, BackendDeveloper):
+    def __init__(self, name):
+        super(FullStackDeveloper, self).__init__(name)
+        self.skills = [
+            "Python",
+            "SQL",
+            "Django",
+            "JavaScript",
+            "HTML",
+            "CSS"
+        ]
```

Comment:
So now it is bad style because you have to add skills that you have added in parents, you repeat yourself.
In every class except `SoftwareEngineer` you __reassign__ attribute `self.skills` that you assign in `SoftwareEngineer`. If you extend `self.skills` instead of __reasign__ - you won't have to assign that big list in `FullstackDeveloper` and you will have this skills just by using super().__ init__.

Reply:
Thank you, changes provided. I didn`t know that FullstackDeveloper will inherit self.skills both from FrontendDeveloper and BackendDeveloper simultanously, just by super(), and self.skills will be sliced. I hope, now i understand super() a bit better.
P.S. I hope, i understood the problem correctly.